### PR TITLE
Implement live JS reload functionality

### PR DESCRIPTION
**Pull request by goshakkk**
This also needs

* a new release of commonjs-require-definition with `require.reset` support
* a new Brunch release with the bumped commonjs-require-definition

### CODE REVIEW

**Comment by goshakkk** [comment]
A sample React/Redux app that will work with this (and the unreleased commonjs, + custom saving `store` in global and `react-proxy` handling) — https://github.com/goshakkk/brunch-livejs-reload-stage1

File: vendor/auto-reload.js [hunk 1]
```diff
@@ -32,6 +32,34 @@
 
       // Hack to force page repaint after 25ms.
       if (forceRepaint) setTimeout(function() { document.body.offsetHeight; }, 25);
+    },
+
+    javascript: function(){
+      console.log(1);
```
Comment (paulmillr) [inline vendor/auto-reload.js]: .

Comment (goshakkk) [inline vendor/auto-reload.js]: Fixed

File: vendor/auto-reload.js [hunk 2]
```diff
@@ -32,6 +32,34 @@
...
+
+      var loaded = 0;
+      var all = srcScripts.length;
+      var onLoad = function() {
+        loaded = loaded + 1;
+        if (loaded === all) {
+          if (typeof require === 'object' && require.reset) require.reset();
+          textScripts.forEach(function(script) { eval(script); });
+        }
+      }
```
Comment (goshakkk) [inline vendor/auto-reload.js]: This is needed to re-evaluate the text scripts that enter the application (like `require('initialize');`)

**Pull request merged**